ql/src/test/queries/clientpositive/antijoin2.q

```diff
@@ -0,0 +1,75 @@
+set hive.merge.nway.joins=false;
+set hive.vectorized.execution.enabled=false;
+set hive.auto.convert.join=true;
+set hive.auto.convert.anti.join=true;
+
+drop table if exists tt1;
+drop table if exists tt2;
+drop table if exists tt3;
+
+create table tt1 (ws_order_number bigint, ws_ext_ship_cost decimal(7, 2));
+create table tt2 (ws_order_number bigint);
+create table tt3 (wr_order_number bigint);
+
+insert into tt1 values (42, 3093.96), (1041, 299.28), (1378, 85.56), (1378, 719.44), (1395, 145.68);
+insert into tt2 values (1378), (1395);
+insert into tt3 values (42), (1041);
+
+-- The result should be the same regardless of vectorization.
+
+explain
```

Comment:
The plans for these queries don't show the pattern of MergeJoin --> MapJoin.  Without this pattern, as I noted in the Jira, the wrong results issue was not reproducible.  In order to force this plan, I had to set the following stats:
alter table tt1 update statistics set ('numRows'='10000000');
alter table tt2 update statistics set ('numRows'='10000000');
alter table tt3 update statistics set ('numRows'='2');
Could you add/modify the existing tests to set these stats and generate the MergeJoin-->MapJoin plan and verify that the fix works for that ?

Reply:
I updated the qfile and expected output for verifying MergeJoin -> MapJoin pattern.

However, I think this issue is unrelated with MergeJoin -> MapJoin pattern and depends only on MapJoinOperator. For example, the second query in this qfile contains only one Join, but the current Hive returns no rows although tt1 - tt2 is not empty. Also, I have checked that the current Hive returns wrong result for the first query without MergeJoin -> MapJoin pattern. (I used mvn test to run both of the queries.)

I think you could reproduce this issue with MapJoin -> MapJoin pattern if you disable vectorized execution. Could you please give me more details about the configuration that you used for reproducing the problem? Also I would be glad if you could run this qfile in your local environment and share the result.

Reply:
We tested locally with the tpcds queries and your patch seems to have fixed the problem.

Thanks!